internal/types/array.go

```diff
@@ -128,3 +132,58 @@ func (a *Array) Append(values ...any) error {
 	a.s = append(a.s, values...)
 	return nil
 }
+
+// HasSameTypeElements returns true if Array elements has the same type.
+// MongoDB consider int32, int64 and float64 that could be converted to int as the same type.
+func (a *Array) HasSameTypeElements() bool {
```

Comment:
That function has much more in common with unexported typeCode type and other type code / alias logic than with anything else in types package. Let's move it to `common` package

Reply:
Moved it to typecode.go and unexported it.